lib/types.go

```diff
@@ -41,6 +42,18 @@ type Options struct {
 	Thresholds map[string][]*Threshold `json:"thresholds"`
 }
 
+type SourceData struct {
```

Comment:
@uppfinnarn Thinking about this, to me it seems more logical that the user tells k6 (via the command line) what type of input data they are supplying. I.e. they should *not* tell k6 what type of runner they want, because that is cognitively harder than just saying what type of data they are giving k6 - users don't know what a "runner" is, and they shouldn't have to know that either. They should say "here is my data, and it is a single URL", or "here is my data, it is a JS script", or "Here is my data, it is a list of URLs". That means command-line options should be parsed to first generate a SrcData struct that includes the source data type, and then that data should be used to determine what type of runner to start. As it is, we use command-line options to determine what runner to start, and then we create the SrcData struct to feed to the runner. Are you OK with changing this?  It means --type gets a slightly different meaning. Because then I think fixing this remaining issue becomes fairly obvious. If not, the code feels like it becomes more convoluted - we will have to use the runner type requested by the user as an input when we try to detect what type of data they are supplying.